### PR TITLE
Sprint 4/issue 21#add functionality for users to add new flight reservations and implement storage in firebase

### LINKED ISSUE #21: Add functionality for users to add new flight reservations and implement storage in Firebase
## DESCRIPTION

Implement functionality for users to input and submit flight reservation details, and save it to firestore.

Develop the necessary backend logic to handle user input for flight details, including origin, destination, date, and number of passengers.
Implement data validation to ensure the entered information is accurate and complete.
Handle any potential errors or edge cases that may arise during the reservation submission process.

### PR DESCRIPTION
**Pull request by Tole15**
Successful Implementation of Flight Reservation Data, Autocomplete Picker and Data Storage

This update signifies the completion and successful operation of the flight reservation country data autocomplete picker feature along with the seamless storage of reservation information. The autocomplete picker has been integrated into the system, offering users an enhanced booking experience by simplifying the input of the country details and flight preferences. It intelligently suggests options as users type, making the process quicker and more convenient.

Furthermore, the implementation of data storage functionality ensures that users' flight reservation details are securely saved and readily accessible. By securely storing this information, users can easily manage their bookings, retrieve important details, and have a hassle-free travel planning experience.

Both the autocomplete picker and the reservation data storage are now fully operational, providing users with a comprehensive and efficient flight reservation system. These updates significantly enhance the overall user experience and contribute to a seamless and reliable flight booking process.

## Screenshots
Screenshot_1689296365
Screenshot_1689296370
Screenshot_1689296373
Screenshot_1689296414
Screenshot_1689296417
Screenshot_1689296359

Co-authored-by: BrandonVG <[brandon.vadillox3@gmail.com](mailto:brandon.vadillox3@gmail.com)>
Co-authored-by: Axw3ll <[axelomar.a.v@gmail.com](mailto:axelomar.a.v@gmail.com)>

### CODE REVIEW

**Review by IvanCa27 — APPROVED** [review]
### _**Good job, guys**_

**Review by Eduardoespl — COMMENTED** [review]
Good Job, lgtm!

**Review by Eduardoespl — APPROVED** [review]

File: src/components/BackHeaderButton.jsx [hunk 1]
```diff
@@ -1,16 +1,22 @@
...
-  </TouchableOpacity>
+const BackHeaderButton = ({previousClick}) => {
+  return (
+    <TouchableOpacity
+      onPress={() => previousClick()}
+      style={BackHeaderButtonStyles.container}>
+      <Ionicons
+        name="airplane-sharp"
+        size={32}
+        color="#5974f5"
```
Comment (YilKanda) [inline src/components/BackHeaderButton.jsx]: If you have a color library, you can use it here as well.

File: src/components/BookingInput.jsx [hunk 2]
```diff
@@ -1,12 +1,37 @@
...
+        clearOnFocus={false}
+        closeOnBlur={true}
+        closeOnSubmit={false}
+        onSelectItem={handleSelect}
+        dataSet={data}
+        textInputProps={{
+          placeholder: 'Select location',
+          style: {backgroundColor: '#FFFFFF'},
+        }}
+        rightButtonsContainerStyle={{backgroundColor: '#FFFFFF'}}
```
Comment (YilKanda) [inline src/components/BookingInput.jsx]: If you have a color library, you can use it here as well.

File: src/components/DatePickerCalendar.jsx [hunk 3]
```diff
@@ -1,94 +1,29 @@
...
-	</View>
-	</View>
-    </Modal>
+    <View style={DatePickerCalendarStyles.modalView}>
+      <DatePicker
+        options={{
+          mainColor: '#6172F6',
+          textFontSize: 17,
+          textHeaderFontSize: 20,
+          selectedTextColor: '#fff',
```
Comment (YilKanda) [inline src/components/DatePickerCalendar.jsx]: If you have a color library, you can use it here as well.

The styles must go in an external file.

File: src/data/Flights.json [hunk 4]
```diff
@@ -1,25 +1,124 @@
...
+      "pais": "Russia"
+    },
+    {
+      "id": "20",
+      "title": "Dubai",
+      "abreviacion": "DXB",
+      "pais": "United Arab Emirates"
+    }
+  ]
+}
```
Comment (YilKanda) [inline src/data/Flights.json]: Add a blank line

File: src/hooks/HandleLogin.js [hunk 5]
```diff
@@ -1,6 +1,20 @@
 import auth from '@react-native-firebase/auth';
 
 const handleLogIn = async (email, password) => {
+  const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+  const passwordRegex =
+    /^(?=.*[A-Za-z])(?=.*\d)(?=.*[@$!%*#?&])[A-Za-z\d@$!%*#?&]{8,}$/;
+
+  if (!emailRegex.test(email)) {
+    console.log('Invalid email format');
```
Comment (YilKanda) [inline src/hooks/HandleLogin.js]: Do not leave debug code in production. You can use Alert.alert() instead.

File: src/hooks/HandleLogin.js [hunk 6]
```diff
@@ -1,6 +1,20 @@
...
+    return {typeError: 'email', message: 'Invalid email format'};
+  }
+
+  if (!passwordRegex.test(password)) {
+    console.log('Incorrect email and/or password');
```
Comment (YilKanda) [inline src/hooks/HandleLogin.js]: Do not leave debug code in production. You can use Alert.alert() instead.

File: src/hooks/SetFlights.js [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+    const user = auth().currentUser;
+    if (user) {
+      const flightsCollection = firestore().collection('flights');
+      await flightsCollection.add({
+        ...flightData,
+        userId: user.uid,
+      });
+    }
+  } catch (error) {
+    console.error('Error al agregar los valores a Firestore:', error);
```
Comment (YilKanda) [inline src/hooks/SetFlights.js]: Do not leave debug code in production. You can use Alert.alert() instead.

File: src/screens/Booking.jsx [hunk 8]
```diff
@@ -1,32 +1,175 @@
...
+            destinationCity,
+            destinationCountry,
+            originCity,
+            originCountry,
+            passengers,
+          });
+          Nextclick();
+          setIsButtonDisabled(false);
+        } catch (error) {
+          console.error(error);
```
Comment (YilKanda) [inline src/screens/Booking.jsx]: Do not leave debug code in production.

File: src/styles/AddButtonStyles.js [hunk 9]
```diff
@@ -0,0 +1,18 @@
+import {StyleSheet} from 'react-native';
+
+const AddButtonStyles = StyleSheet.create({
+  button: {
+    backgroundColor: '#5974f5',
```
Comment (YilKanda) [inline src/styles/AddButtonStyles.js]: If you have a color library use it.

File: src/styles/BookingInputStyles.js [hunk 10]
```diff
@@ -1,20 +1,12 @@
-import { StyleSheet } from 'react-native';
+import {StyleSheet} from 'react-native';
 
 const BookingInputStyles = StyleSheet.create({
-  container:{
-    marginTop:90,
+  container: {
+    marginTop: 90,
     marginLeft: 21,
     borderBottomColor: '#bcbcbc',
```
Comment (YilKanda) [inline src/styles/BookingInputStyles.js]: If you have a color library use it.

File: src/styles/BookingStyles.js [hunk 11]
```diff
@@ -1,16 +1,24 @@
-import { StyleSheet } from "react-native";
+import {StyleSheet} from 'react-native';
 
 const BookingStyles = StyleSheet.create({
   screenContainer: {
-    flex:1,
-    backgroundColor:'white',
+    flex: 1,
+    backgroundColor: 'white',
```
Comment (YilKanda) [inline src/styles/BookingStyles.js]: If you have a color library use it.

File: src/styles/NextButtonStyles.js [hunk 12]
```diff
@@ -1,29 +1,29 @@
...
+    backgroundColor: '#5974f5',
+    paddingVertical: 11,
+    paddingHorizontal: 10,
+    width: '90%',
+    borderRadius: 8,
+    elevation: 7,
+    shadowColor: '#0000ff',
+    shadowOffset: {width: 0, height: 2},
+    shadowOpacity: 0.3,
+    shadowRadius: 2,
```
Comment (YilKanda) [inline src/styles/NextButtonStyles.js]: Sort the styles alphabetically.

File: src/styles/NextButtonStyles.js [hunk 13]
```diff
@@ -1,29 +1,29 @@
...
+  },
+  enabledBtn: {
+    backgroundColor: '#5974f5',
+  },
+  disabledBtn: {
+    backgroundColor: '#9e9e9e',
+  },
+  buttonText: {
+    color: '#fbfbfc',
```
Comment (YilKanda) [inline src/styles/NextButtonStyles.js]: If you have a color library, use it.

File: src/styles/NextButtonStyles.js [hunk 14]
```diff
@@ -1,29 +1,29 @@
...
+    textAlign: 'center',
+    fontSize: 19,
+    fontWeight: 'bold',
```
Comment (YilKanda) [inline src/styles/NextButtonStyles.js]: Sort your styles alphabetically

**Review by YilKanda — CHANGES_REQUESTED** [review]

**Pull request merged**